cfme/utils/log_validator.py

```diff
@@ -46,16 +46,23 @@ def __init__(self, remote_filename, **kwargs):
         self._remote_file_tail = SSHTail(remote_filename, **kwargs)
         self.matches = {}
 
+    def update_matched_patterns(self, new_patterns):
```

Comment:
A docstring would be nice to have here. We need to elaborate our usecase and reason why we implemented this. So please add a comment telling this function is used when periodically probing the log for a list of patterns where we want to shrink the list when targets are found.